tst/com/amazon/corretto/crypto/provider/test/EcGenTest.java

```diff
@@ -325,9 +325,12 @@ public void threadStorm() throws Throwable {
 
     public static void assertECEquals(final String message, final ECPrivateKey expected,
                                        final ECPrivateKey actual) {
+        final boolean expectEncodingCompat = TestUtil.getJavaVersion() != 10;
```

Comment:
Could the implementation of `getEncoded` in Java 10 be switched in future or other versions decide to do similar things? If yes, in the test code, could we trim the zeros regardless of the Java version and always compare? It would be nice to make the tests not rely on implementation details so that if other versions of Java decide to trim, these tests would not fail.

Reply:
no, Java 10 is end-of-life and hasn't even taken security patches in years (hence the bug we're seeing) so i don't think its `getEncoded` implementation will change.

if other versions of java do decide to re-introduce the trimming bug, i'd rather our tests fail. if we're made aware of the issue by our tests failing, we can submit a fix upstream to OpenJDK before the affected version is EOL'd and cast into carbonite like JDK10.